### ISSUE TITLE
auth.service.js problem with isAdmin() and hasRole(...) function

### ISSUE THREAD

**Issue by Emidomenge** [angular-fullstack/generator-angular-fullstack#1976]
By default an user can be` 'guest' `or` 'user' `or `'admin'`.

I added 2 new roles in my project such as` 'artist' `or` 'subscriber'`. 
Now server\config\environment\ **shared.js** looks like this  : 
`userRoles: ['guest', 'user', 'admin', 'artist','subscriber']`

When I set one of those 2 role to my user, I have some issues using  isAdmin() and hasRole(...) function which are in : `client/components/auth/auth.service.js`

For example, on my navbar, I'm not able to hide 'Admin' tab using isAdmin() : 
`<li ng-show="nav.isAdmin()" ui-sref-active="active"><a ui-sref="admin">Admin</a></li>`

nav.isAdmin() return true, always. But it returns correctly false if `user.role = 'guest' or 'user' `so maybe I'm missing something ? ...

I tried also : 
`<li ng-show="nav.hasRole('admin')" ui-sref-active="active"><a ui-sref="admin">Admin</a></li>`
and in my controller : `this.hasRole = Auth.hasRole;`
It returns also true... which should not be the case !

Any suggestion ? Thanks !
 - [x] I understand that GitHub issues are not for tech support, but for questions specific to this generator, bug reports, and feature requests.

Item | Version
----- | -----
generator-angular-fullstack | 3.6.1
Node | 5.7.1
npm | 3.6.0
Operating System |  Windows 7
etc | etc

Item | Answer
----- | -----
Transpiler | Babel 
Markup | HTML
CSS | CSS 
Router | ngRoute 
Build Tool | Grunt
Client Tests | Jasmine 
DB | MongoDB 
Auth | Y

**Comment by Awk34** [comment]
So if you look at how role checking is done one the [client]() and the [server](), you'll see that it's not an equality comparison. Instead, it checks if the user's role is greather than or equal to the requested role. That was put in so that admins will still have the permissions of a normal user, on top of admin permissions. I realize that the wording of the functions is confusing. You'd expect `hasRole` do be an equality comparison, and maybe `hasRoleOrGreater` to be how it works currently.

**Comment by Awk34** [comment]
tl;dr with how it works now, your array should be `['guest', 'user', 'artist','subscriber', 'admin']` (always keep admin at the end)

**Comment by Emidomenge** [comment]
Indeed it was a little bit confusing for me. Now it works as expected, thanks !
